### PR TITLE
Create API to check if site has initial publish

### PR DESCRIPTION
**Pull request by phuongnq**
### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/5847

```
GET /studio/api/2/publish/has_initial_publish?siteId=ed
{
    "response": {
        "code": 0,
        "message": "OK",
        "remedialAction": "",
        "documentationUrl": ""
    },
    "hasInitialPublish": true
}
```

### CODE REVIEW

File: src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java [hunk 1]
```diff
@@ -215,6 +207,23 @@ public ResponseBody getAvailablePublishingTargets(@RequestParam(name = REQUEST_P
         return responseBody;
     }
 
+    @ValidateParams
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java]: @ValidateParams annotation has been removed. Please annotate the Controller class @Validated (from org.springframework.validation.annotation)

File: src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java [hunk 2]
```diff
@@ -215,6 +207,23 @@ public ResponseBody getAvailablePublishingTargets(@RequestParam(name = REQUEST_P
...
+    @GetMapping(value = HAS_INITIAL_PUBLISH, produces = APPLICATION_JSON_VALUE)
+    public ResponseBody hasInitialPublish(@EsapiValidatedParam(type = SITE_ID) @RequestParam(name = REQUEST_PARAM_SITEID) String siteId)
+        throws SiteNotFoundException {
+        if (!siteService.exists(siteId)) {
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java]: Let's throw these exceptions from the service layer
Let's use
```siteService.checkSiteExists(siteId);```

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java (hunk 1 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java]: Updated to use `@Validated` and `@Valid`

File: src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java]: Moved check site exists to services layer. Thanks.

**Review by jmendeza — APPROVED** [review]

**Pull request merged**